test/cipher_test.rb

```diff
@@ -75,30 +82,58 @@ def test_it_can_decrypt_whole_message
     assert_equal "hello world", @cipher.decrypt_message("keder ohulw", expected)
   end
 
+  def test_it_can_split_return_value_of_generate_random_key_into_separate_keys
+    expected = [02, 27, 71, 15]
+    assert_equal expected, @cipher.shift_generator.key_gen.create_key("02715")
+  end
+
   def test_it_can_verify_key
     assert_equal true, @cipher.shift_generator.key_gen.verify_key("02715")
     assert_equal false, @cipher.shift_generator.key_gen.verify_key("999999")
   end
 
+  def test_it_can_get_last_4_digits_from_sqaure_date
+    squared_date_param = "73018848400"
+    assert_equal "8400", @cipher.shift_generator.offset_gen.last_four_digits(squared_date_param)
+  end
+
+  def test_it_can_separate_last_4_digits_from_square_dates_into_keys
+    four_digits = "8400"
+    expected = [8, 4, 0, 0]
+    assert_equal expected, @cipher.shift_generator.offset_gen.create_offset(four_digits)
+  end
+
+  def test_it_can_get_final_shift_values
+    expected =  [10, 31, 71, 15]
+    assert_equal expected, @cipher.shift_generator.shift_key("02715", "260220")
+  end
+
+  def test_it_can_separate_last_4_digits_from_square_dates_into_keys
+    four_digits = "8400"
+    expected = [8, 4, 0, 0]
+    assert_equal expected, @cipher.shift_generator.offset_gen.create_offset(four_digits)
+  end
+
```

Comment:
added a lot of tests to cipher to make sure that everything housed in cipher was being tested